### PR TITLE
created feature/LineItem-entity

### PR DESCRIPTION
**Pull request by thebaker10**

### CODE REVIEW

File: docker/api/src/Domain/LineItem/LineItem.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use JetBrains\PhpStorm\ArrayShape;
+use JsonSerializable;
+
+#[Entity, Table(name: 'lineItem')]
```
Comment (flierc-csp) [inline docker/api/src/Domain/LineItem/LineItem.php]: Should this be `line_item` rather than `lineItem`?

**Review by flierc-csp — CHANGES_REQUESTED** [review]

File: docker/api/src/Application/Actions/LineItem/LineItemPostAction.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    protected function action(): Response
+    {
+
+        $payload = $this->request->getParsedBody();
+        $orderId = $payload['orderId'];
+        $productId = $payload['productId'];
+        $lineItemPrice = $payload['lineItemPrice'];
+        $quantity = $payload['quantity'];
+        $lineItem = new Line_Item($orderId, $productId, $lineItemPrice, $quantity);
+
```
Comment (noko12312) [inline docker/api/src/Application/Actions/LineItem/LineItemPostAction.php:52]:
```suggestion
$product = $this->em->getRepository(Product::class)->find($productId);
$lineItem->setProduct($product);
```

File: docker/api/src/Domain/LineItem/LineItem.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    #[Column(type: 'integer', unique: false, nullable: false)]
+    private string $quantity;
+
+    #[ManyToOne(targetEntity: Product::class)]
+    #[JoinColumn(name: 'product_id', referencedColumnName: 'product_id')]
+    private Product $product;
+
+    // #[ManyToOne(targetEntity: Order::class)]
+
```
Comment (noko12312) [inline docker/api/src/Domain/LineItem/LineItem.php]:
```suggestion
    public function setProduct(Product $product): void{
        $this->product = $product;
    }
```

**Review by noko12312 — CHANGES_REQUESTED** [review]

File: docker/api/src/Domain/LineItem/LineItem.php (hunk 3 above)
Comment (noko12312) [reply docker/api/src/Domain/LineItem/LineItem.php]: Without setting the product object, you will encounter an error "product_id cannot be null" when trying to add to database.  It seems all ManyToOne associations will throw this error, so the order object will also need to be set LineItemPostAction when the Order entity is completed.

Comment (flierc-csp) [reply docker/api/src/Domain/LineItem/LineItem.php]: Great catch!  I already forgot about that.

**Review by noko12312 — APPROVED** [review]

**Pull request merged**